src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1165,3 +1165,36 @@ def handle_version_update():
             refresh_known_clouds()
     except Exception as ex:  # pylint: disable=broad-except
         logger.warning(ex)
+
+
+def resource_to_scopes(resource):
+    """Convert the ADAL resource ID to MSAL scopes by appending the /.default suffix and return a list.
+    For example: 'https://management.core.windows.net/' -> ['https://management.core.windows.net/.default']
+    :param resource: The ADAL resource ID
+    :return: A list of scopes
+    """
+    if 'datalake' in resource or 'batch' in resource or 'database' in resource:
```

Comment:
Is it possible that we move this tricky logic from core to module level?

Reply:
Good point. Currently `resource_to_scopes` is only called by `_get_mgmt_service_client`. We should actually remove this tricky logic totally from this function. Will do in a separate PR.